account_invoice_fiscal_position_update/models/account_invoice.py

```diff
@@ -40,6 +40,12 @@ def fiscal_position_change(self):
                 else:
                     taxes = product.supplier_taxes_id.filtered(
                         lambda tax: tax.company_id == self.company_id)
+                    # If you change from purchase line the taxes, maintain it.
+                    if hasattr(line.invoice_id, 'purchase_id') and \
+                            line.invoice_id.purchase_id and \
+                            line.purchase_line_id and \
+                            line.purchase_line_id.taxes_id:
+                        taxes = line.purchase_line_id.taxes_id
```

Comment:
Clarify hack + remove extra useless condition
```suggestion
                    # HACK: using hasattr for not depending on purchase
                    if (hasattr(line.invoice_id, 'purchase_id') and
                            line.invoice_id.purchase_id and
                            line.purchase_line_id.taxes_id):
                        taxes = line.purchase_line_id.taxes_id
```